src/utils/withWidth.js

```diff
@@ -22,8 +22,11 @@ export default function withWidth(options = {}) {
         width: SMALL,
       };
 
-      componentDidMount() {
-        this.updateWidth();
+      componentWillMount() {
```

Comment:
Thanks for raising this issue.
I think that we should change our approach. One other issue I have with the current solution is that we might have a visual blink on desktop.
What about not rendering the element when doing SSR and waiting for the js browser engin to be available to actually render the child element?